code/Forms/GridFieldDetailFormPreviewExtension.php

```diff
@@ -31,6 +33,24 @@ protected function updateItemEditForm(Form $form): void
         }
     }
 
+    protected function updateFormActions(FieldList $actions): void
```

Comment:
This is not appropriate to put on `GridFieldDetailFormPreviewExtension`. Someone might remove this extension because they don't want the preview stuff happening (or extend it or whatever) - but that functionality has nothing to do with the unsaved changes indicator. This should be in a separate extension.

Reply:
If these should be combined in a major release when we can rename the extension to something that isn't explicitly about preview, we can open an issue to do that.